inc/Ui/MediaDiff.php

```diff
@@ -0,0 +1,339 @@
+<?php
+
+namespace dokuwiki\Ui;
+
+use dokuwiki\ChangeLog\MediaChangeLog;
+use dokuwiki\Ui\MediaRevisions;
+use dokuwiki\Extension\Event;
+use dokuwiki\Form\Form;
+use JpegMeta;
+
+/**
+ * DokuWiki MediaDiff Interface
+ *
+ * @package dokuwiki\Ui
+ */
+class MediaDiff extends Diff
+{
+    /* @var MediaChangeLog */
+    protected $changelog;
+
+    /**
+     * MediaDiff Ui constructor
+     *
+     * @param string $id  media id
+     */
+    public function __construct($id)
+    {
+        if ($id) {
+            throw new \InvalidArgumentException('media id should not be empty!');
+        }
+        $this->item = 'media';
+
+        // init preference
+        $this->preference['fromAjax'] = false; // see doluwiki\Ajax::callMediadiff()
+        $this->preference['showIntro'] = false;
+        $this->preference['difftype'] = 'both';  // media diff view type: both, opacity or portions
+
+        parent::__construct($id);
+    }
+
+    /** @inheritdoc */
+    protected function setChangeLog()
+    {
+        $this->changelog = new MediaChangeLog($this->id);
+    }
+
+    /** @inheritdoc */
+    protected function itemFN($id, $rev = '')
+    {
+        return mediaFN($id, $rev);
+    }
+
+    /** @inheritdoc */
+    protected function preProcess()
+    {
+        parent::preProcess();
+        if (!isset($this->oldRev, $this->newRev)) {
+            // no revision was given, compare previous to current
+            $revs = $this->changelog->getRevisions(0, 1);
+            $this->oldRev = file_exists(mediaFN($this->id, $revs[0])) ? $revs[0] : '';
+            $this->newRev = '';
+        }
+    }
+
+    /**
+     * Shows difference between two revisions of media
+     *
+     * @author Kate Arzamastseva <pshns@ukr.net>
+     */
+    public function show()
+    {
+        global $conf;
+
+        $ns = getNS($this->id);
+        $auth = auth_quickaclcheck("$ns:*");
+
+        if ($auth < AUTH_READ || !$this->id || !$conf['mediarevisions']) return '';
+
+       // determine left and right revision
+        if (!isset($this->oldRev, $this->newRev)) $this->preProcess();
+        [$oldRev, $newRev] = [$this->oldRev, $this->newRev];
+
+        // prepare event data
+        // NOTE: MEDIA_DIFF event does not found in DokuWiki Event List?
+        $data = array();
+        $data[0] = $this->id;
+        $data[1] = $oldRev;
+        $data[2] = $newRev;
+        $data[3] = $ns;
+        $data[4] = $auth; // permission level
+        $data[5] = $this->preference['fromAjax'];
+
+        // trigger event
+        Event::createAndTrigger('MEDIA_DIFF', $data, null, false);
```

Comment:
**MEDIA_DIFF** event is not documented in [devel:events_list](https://www.dokuwiki.org/devel:events_list) and not used in any plugins. I feel like to remove this event.

Reply:
**MEDIA_DIFF** event has removed in 629a763